servicex/web/servicex_file.py

```diff
@@ -22,11 +24,37 @@ def servicex_file():
     backend_type = matches[0]
     sub = session.get('sub')
     user = UserModel.find_by_sub(sub)
+
+    endpoint_url = get_correct_url(request)
     body = f"""\
     api_endpoints:
-      - endpoint: {request.url_root}
+      - endpoint: {endpoint_url}
         token: {user.refresh_token}
         type: {backend_type}
     """
     return send_file(BytesIO(dedent(body).encode()), mimetype="text/plain",
                      as_attachment=True, attachment_filename="servicex.yaml")
+
+
+def get_correct_url(request: flask.Request) -> str:
+    """
+    Update url string to try to make sure that the proper url scheme (https, http)
+    is used
+    see https://github.com/ssl-hep/ServiceX/issues/266
+
+    :param request: flask request object
+    :return: string with http url changed to https except
+    """
+
+    parsed_url = urlparse(request.base_url)
+    request_scheme = request.headers.get('X-Scheme')
+    if request_scheme is not None:
+        # use the same scheme that the request used
+        return parsed_url._replace(scheme=request_scheme).geturl()
+    elif parsed_url.scheme == "http" and "localhost" not in parsed_url.netloc:
+        # if the request scheme is unknown use https unless we're referring
+        # to localhost
+        return parsed_url._replace(scheme="https").geturl()
```

Comment:
Not sure we should be relying on a private method here?

Reply:
I'm kind of up in the air on this.  The _replace method is documented on the collections page and returns a new instance with the change implemented so I'm not sure it's a typical private method.  The parsed_url is a namedtuple so it's immutable.